app/assets/javascripts/controllers/schedules/controller.js

```diff
@@ -8,6 +8,44 @@
 Yacs.views.schedules = function (target, params) {
   Yacs.render(target, 'schedules');
 
+  var schedule_ids = [];
+
+  // check for query parameters
+  var query = window.location.hash.split('?')[1]
+  if(typeof query !== 'string' || query.length < 1) {
+    /* If there are no query parameters, make the current
+     * selected section ids into a query parameter and
+     * re-navigate the page to that.
+     * This is so someone who clicks on the Schedule button
+     * after selecting courses will get the copiable link in
+     * their address bar.
+     */
+    Yacs.router.visit('/schedules?section_ids='+ Yacs.user.getSelectionsRaw());
```

Comment:
My reasoning for using the URL is that people's instinct and the most straightforward thing to do is to copy the address bar. Adding a share button isn't as intuitive, and it's not bookmarkable (and I think a lot of people would start using bookmarks to schedules if they could).

I am going to make it so that the query will write into the selection if nothing is selected, yes.